web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/VariantsApiController.java

```diff
@@ -43,32 +42,19 @@
         HttpStatus httpStatus = HttpStatus.OK;
         List<Alteration> alterationList = new ArrayList<>();
         if (hugoSymbol != null || entrezGeneId != null) {
-            Gene gene = null;
-            if (entrezGeneId != null && hugoSymbol != null && !GeneUtils.isSameGene(entrezGeneId, hugoSymbol)) {
-                httpStatus = HttpStatus.BAD_REQUEST;
-            } else {
-                if (entrezGeneId != null) {
-                    gene = GeneUtils.getGeneByEntrezId(entrezGeneId);
-                }
-
-                if (hugoSymbol != null && gene == null) {
-                    gene = GeneUtils.getGeneByHugoSymbol(hugoSymbol);
-                }
-
-                if (gene != null) {
-
-                    Set<Alteration> allAlterations = AlterationUtils.getAllAlterations(gene);
-                    if (variant == null && proteinStart == null && proteinEnd == null) {
-                        alterationList.addAll(allAlterations);
-                    } else {
-                        AlterationBo alterationBo = new ApplicationContextSingleton().getAlterationBo();
-                        Alteration alteration = AlterationUtils.getAlteration(gene.getHugoSymbol(), variant, variantType, consequence, proteinStart, proteinEnd);
-                        alterationList.addAll(alterationBo.findRelevantAlterations(alteration, new ArrayList<Alteration>(AlterationUtils.getAllAlterations(gene))));
-                    }
+            Gene gene = GeneUtils.getGene(entrezGeneId, hugoSymbol);
```

Comment:
I dont think I need to check same gene issue anymore. EntrezGeneId has higher priority then hugoSymbol. It will be solved with getGene function